kura/org.eclipse.kura.container.orchestration.provider/src/main/java/org/eclipse/kura/container/orchestration/provider/DockerService.java

```diff
@@ -22,34 +22,44 @@
  */
 public interface DockerService {
 
-    /**
-     * Lists all available containers, like running 'Docker ps -a'
-     *
-     * @return
-     */
-    public String[] listContainers();
-
     /**
      * Lists all available containers by id only, like running 'Docker ps -a'
      *
      * @return
      */
-    public String[] listContainersByID();
+    public List<String> listContainersIds();
 
     /**
      * Lists all running containers by string of ID's
      *
      * @return
      */
-    public ContainerDescriptor[] listByContainerDescriptor();
+    public List<ContainerDescriptor> listContainerDescriptors();
+    
+    /**
+    *
+    * @param imageName
+    * @throws KuraException
+    */
+   public void pullImage(String imageName, String imageTag) throws KuraException;
+
+   /**
+    *
+    * @param imageName
+    * @param timeOutSecconds
+    * @throws KuraException
+    */
+   public void pullImage(String imageName, String imageTag, int timeOutSecconds) throws KuraException;
```

Comment:
Going with the theme of my previous comment above, perhaps we add all of the timeOut Parameters into the ContainerDescriptor?

Reply:
Agreed. Can you do that?